template/apps/api/src/utils/object.util.ts

```diff
@@ -0,0 +1,10 @@
+import _ from 'lodash';
+
+const flattenObject = <T>(obj: object, path: string[] = []): Record<string, T[keyof T]> =>
+  !_.isObject(obj)
+    ? { [path.join('.')]: obj }
+    : Object.entries(obj).reduce((cur, [key, value]) => _.merge(cur, flattenObject<T>(value, [...path, key])), {});
```

Comment:
flip this condition. More readable case when we don't need to apply logical not to the condition.